### PR TITLE
fix: work on node + browser

### LINKED ISSUE #35: .js in import of ESM
I am using Yarn to download this package:
```
yarn add @material/material-color-utilities
```

After fixing #29 I got these error:
```
$ C:\files\src\hct\node_modules\.bin\ts-node-esm .\index.ts
C:\files\src\hct\node_modules\ts-node\dist-raw\node-esm-resolve-implementation.js:383
    throw new ERR_MODULE_NOT_FOUND(
          ^
CustomError: Cannot find module 'C:\files\src\hct\node_modules\@material\material-color-utilities\dist\blend\blend' imported from C:\files\src\hct\node_modules\@material\material-color-utilities\dist\index.js
    at finalizeResolution (C:\files\src\hct\node_modules\ts-node\dist-raw\node-esm-resolve-implementation.js:383:11)
    at moduleResolve (C:\files\src\hct\node_modules\ts-node\dist-raw\node-esm-resolve-implementation.js:818:10)
    at Object.defaultResolve (C:\files\src\hct\node_modules\ts-node\dist-raw\node-esm-resolve-implementation.js:929:11)
    at C:\files\src\hct\node_modules\ts-node\src\esm.ts:228:33
    at entrypointFallback (C:\files\src\hct\node_modules\ts-node\src\esm.ts:179:34)
    at resolve (C:\files\src\hct\node_modules\ts-node\src\esm.ts:227:12)
    at resolve (C:\files\src\hct\node_modules\ts-node\src\child\child-loader.ts:19:39)
    at ESMLoader.resolve (node:internal/modules/esm/loader:580:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:294:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:80:40)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

When I checking `dist\index.js`:
```typescript
export * from './blend/blend';
```

I made some changes to this:
```typescript
export * from './blend/blend.js';
```

After I adding `.js` to all imports. It works perfect to me.

More info: [export - JavaScript | MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export#using_export_from)

### LINKED ISSUE #68: Use with latest Node.JS does NOT work because imports and exports do not follow ES6 modules rules, need file extension on import path.
Hello I tried to use this module with Next.JS and it was returning the 'ERR_MODULE_NOT_FOUND' when trying to use it. To fix the issue I added the ".js" extension to all the exports and imports on all files.

### PR DESCRIPTION
**Pull request by WillsterJohnson**
Currently the only working version of this lib is [`@importantimport/material-color-utilities`](https://github.com/importantimport/material-color-utilities)

---

- [x] passes all test cases
- [x] TypeScript builds successfully
- [x] does not break existing behavior

Fixes #35 
Fixes #68 

This PR allows the NPM package to work on all modern browsers and on Node, where currently it does not work anywhere.

To maintain the correct behavior of the JS code being able to run in JS environments, ensure that all import statements point to a `*.js` file.

Recommended version bump: Minor
Reason: adds new functionality;
- compatibility with Node.js (LTS, 16+)
- compatibility with Gecko browser (FireFox 60+)
- compatibility with Chromium browser (Chrome 61+)
- compatibility with WebKit browser (Safari 11+)

### CODE REVIEW

**Comment by willster277** [comment]
The issue fixed by this PR was first created in [this commit](https://github.com/material-foundation/material-color-utilities/tree/bd6151cf74b79953aa73d3260289b870c08113d1), meaning today marks 440 (four hundred forty) days of the TypeScript library being defunct on all JavaScript runtime environments.

See: #35, #68 

---

@material-admin,  without this PR, the TypeScript library **does not work**
@kluever @guidezpl

**Review by Reallukeisbest — APPROVED** [review]

**Comment by kwaa** [comment]
It seems that Google apparently doesn't care much about the availability of this library...
or they ship TypeScript directly internally.

**Review by kwaa — APPROVED** [review]

**Comment by WillsterJohnson** [comment]
@kwaa It's quite common of Google to do this. Create something good (the Material You design system), then take a big ol' dump on it (this 467-day old issue rendering the library useless).
Such a shame, if they fixed these problems with their softwares they would probably push MS and Apple into obscurity.

**Comment by rodydavis** [comment]
LGTM

**Comment by WillsterJohnson** [comment]
pulled updates to the fork & fixed newly introduced fatal syntax errors. @rodydavis , please review these changes if you have any concerns.

As this repo is a mirror, here is a bash command which will automatically apply the changes this PR makes;

```sh
echo "const f=require('fs');const p=require('path');const x=(d)=>{for(const t of f.readdirSync(d)){if(t.endsWith('.ts'))f.writeFileSync(p.join(d,t),f.readFileSync(p.join(d,t),'utf8').replace(/(im|ex)port(.+from.+?'.+)(?<!(\.js|jasmine))';/g,'$1port$2.js\';'),'utf8');else if(f.statSync(p.join(d,t)).isDirectory())x(p.join(d,t))}};x(process.cwd());" | node
```

If it turns out that the Material Foundation dev team for whatever reason won't include the `.js` extension that is **required** for ESM code to execute in a JavaScript runtime environment, then it would be a good idea to put in place some kind of enforcement.

Be it a github workflow

```yml
name: Fix fatal coding practice

on:
  push:
    branches:
      - "main"

jobs:
  fixfatalerrors:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Setup Node
        uses: actions/setup-node@v3
        with:
          node-version: "18.x"
          registry-url: "https://registry.npmjs.org"
      
      - name: Fix Fatal Issues
        run: echo "const f=require('fs');const p=require('path');const x=(d)=>{for(const t of f.readdirSync(d)){if(t.endsWith('.ts'))f.writeFileSync(p.join(d,t),f.readFileSync(p.join(d,t),'utf8').replace(/(im|ex)port(.+from.+?'.+)(?<!(\.js|jasmine))';/g,'$1port$2.js\';'),'utf8');else if(f.statSync(p.join(d,t)).isDirectory())x(p.join(d,t))}};x(process.cwd());" | node
```

Or a pre-commit hook

```sh
#!/usr/bin/env sh

git stash

echo "const f=require('fs');const p=require('path');const x=(d)=>{for(const t of f.readdirSync(d)){if(t.endsWith('.ts'))f.writeFileSync(p.join(d,t),f.readFileSync(p.join(d,t),'utf8').replace(/(im|ex)port(.+from.+?'.+)(?<!(\.js|jasmine))';/g,'$1port$2.js\';'),'utf8');else if(f.statSync(p.join(d,t)).isDirectory())x(p.join(d,t))}};x(process.cwd());" | node

git add .

git stash pop

echo "\x1b[35mFIXED FATAL ISSUES IN THE CODEBASE: SEE STAGED CHANGES"
```

Or anything else. The key thing here is for the last 476 days (or longer) the typescript shipped from this repo (potentially others) has been categorically unusable and defunct, and that ***desperately*** needs to change.

Material design is, in my opinion, the best design system by a long shot. I, and many others, would very much like it if the tools made available to us did not cause our applications to crash immediately upon startup, or worse, at any point during runtime.

I hope I've made it clear how frustrating it has been to not be able to use something I believe to be such high quality due to such a basic syntactic error.

---

It would surprise me very much if this repo is the only one effected by this issue, particularly as the same developers working on the TypeScript here are almost certainly working on TypeScript elsewhere. It may be a good idea to check basically every TypeScript file belonging to google that the contributors to this project have been involved with as the same fatal error may be present.
As I said recently, if Google & it's child organisations fixed all these little errors (for context, the JS line, workflow, and pre-commit hook took me a combined total of 30 minutes to create, writing a `.js` extension in the first place is *automated by every text editor unless you go out of your way to disable that*) I'm certain that Google would utterly dominate every digital space it entered.

**Comment by mateoroldos** [comment]
+1 🙏 @kluever

**Comment by pennzht** [comment]
Thank you for your PR. I'm taking a look.

**Comment by pennzht** [comment]
Thank you for your PR!

While we currently don't accept external contributions, we appreciate that you have brought this to our attention.

We have made this change internally as https://github.com/material-foundation/material-color-utilities/commit/41cf95d23acb9c1a9248257dc4571e27c502f795 (also extending the change to newly added files).

We are closing this PR and related issues. Thank you for your support.

File: typescript/quantize/lab_point_provider.ts [hunk 1]
```diff
@@ -15,9 +15,9 @@
  * limitations under the License.
  */
 
-import './point_provider';
+import './point_provider.js';
```
Comment (WillsterJohnson) [inline typescript/quantize/lab_point_provider.ts:18]: typescript/quantize/lab_point_provider.ts:18 - `.js` extension added in this PR but not added in internal change.

**Comment by rodydavis** [comment]
Will update the internal script. Looks like it skipped this file

**Comment by WillsterJohnson** [comment]
Awesome, I guess #82 will be short-lived. I'll still go ahead and make a list of misses, maybe this was the edge case, maybe the update to the internal script will have new ones. Better safe than sorry.

**Pull request closed without merging**